includes/Services/FlowService.php

```diff
@@ -0,0 +1,203 @@
+					// For Indexed Arrays having Non Associative Array Values
+					(!is_array($index_value) && isset($flow_data[$key]))? 
+						$updated_flow_data[$key] = $flow_data[$key] 
+						: $updated_flow_data[$key] = $value;
+				}
+				continue;
+			} 
+
+			// To handle Associative Arrays gracefully
+			$updated_flow_data[$key] = self::update_flow_data_recursive($value, $flow_data[$key]);
+		}
+		return $updated_flow_data;
+	}
+
+	/*
+	 * function to update the Database recursively based on Values opted or entered by the User
+	 */	
+	private static function update_post_call_data_recursive(&$flow_data, $params) {
+		$exception_list = Flows::get_exception_list();
+
+		foreach ($flow_data as $key => $value)
+		{ 		
+			if (! array_key_exists($key, $params)) {
+				$flow_data[$key] = $value;	
+				continue;
+			}
+
+			// Accepts the value of Exception List keys as is from the database
+			if($exception_list && isset($exception_list[$key])) {
+				$flow_data[$key] = $params[$key];
+				continue;
+			}
+
+			// Error thrown if the datatype of the parameter does not match
+			if(strcmp(gettype($value), gettype($params[$key])) != 0) {
+				return new \WP_Error(
+					'wrong_param_type_provided',
+					"Wrong Parameter Type Provided : ". $key. ' => '. gettype($params[$key]) . '. Expected: ' . gettype($value),
+					array( 'status' => 404 )
+				);
+			}
+
+			// Accepts non-Array Values entered by the user
+			if(!is_array($value)) {
+				$flow_data[$key] = $params[$key];
+				continue;
+			}
+
+			// To handle Indexed Arrays gracefully
+			if (count(array_filter(array_keys($params[$key]), 'is_string')) === 0 ) {
+				// If the Database value is empty or an Indexed Array, to avoid Associative arrays to be overwritten (Eg: data)
+				((sizeof($value) === 0) || (count(array_filter(array_keys($value), 'is_string')) === 0))?
+					$flow_data[$key] = $params[$key]
+					: $flow_data[$key] = $value;
+				continue;
+			}
+
+			// To handle Associative Arrays gracefully
+			$flow_data[$key] = self::update_post_call_data_recursive($value, $params[$key]);
+			if( \is_wp_error($flow_data[$key]) ) {
+				return $flow_data[$key];
+			}
+		}					
+		return $flow_data;
+	}
+
+    /*
+	 * Read onboarding flow options from database
+	*/
+	public static function read_flow_data_from_wp_option() {
+		return \get_option( Options::get_option_name( 'flow' ), false );
+	}
+
+    /**
+	 * check the current flow type and update default data if flowtype is ecommerce.
+	 *
+	 * @param default flow data.
+	 *
+	 * @return array
+	 */
+	private static function update_default_data_for_ecommerce( $data ) {
+		// update default data with ecommerce data
+		$data['data']['topPriority']['priority1'] = 'selling';
+		$data['data']['siteType']['referTo'] = 'business';
+		return $data;
+	}
+
+    /*
+	 * function to search for key in array recursively with case sensitive exact match
+	 */
+	public static function check_key_in_nested_array( $params, $flow_data, $header_key = 'Base Level' ) {
```

Comment:
`question` What does the function return if there is no mismatch? The reason I am asking this is because if this is checking for mismatch then this can be named to something more appropriate like `find_mismatch`?